examples/wire_read_tcp.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/python
+import argparse
+import sys
+
+import dns.exception
+import dns.message
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Read sequence of DNS wire formats prefixed with 2-byte "
+        "length field - like from TCP socket - and print the messages. This "
+        "format is used e.g. by dnsperf -B option and dnsgen."
+    )
+    parser.add_argument("infile")
+    args = parser.parse_args()
+    ok_msgs = 0
+    bad_msgs = 0
+
+    with open(args.infile, "rb") as infile:
+        while True:
+            offset = infile.tell()
+            len_wire = infile.read(2)
+            if len(len_wire) == 0:  # end of file - expected
+                break
+            if len(len_wire) == 1:
+                raise ValueError("incomplete length preambule, offset", offset)
```

Comment:
I think you mean `preamble`

Reply:
Indeed. Spelling from my native language leaked into this message :-)

Reply:
No problem!  I would have just fixed it later if I hadn't been making the other comment.